### PR TITLE
i#X Don't close STDERR when closing thread logs

### PR DESCRIPTION
**Pull request by Carrotman42**
Found this during attach/detach testing, but it's not really related to
attach/detach.

### CODE REVIEW

File: core/dynamo.c [hunk 1]
```diff
@@ -2571,7 +2571,7 @@ dynamo_thread_exit_common(dcontext_t *dcontext, thread_id_t id,
 #endif
 
 #ifdef DEBUG
-    if (dcontext->logfile != INVALID_FILE) {
+    if (dcontext->logfile != INVALID_FILE && dcontext->logfile != STDERR) {
```
Comment (derekbruening) [inline core/dynamo.c]: The same check should be done when closing main_logfile, right?

Comment (derekbruening) [inline core/dynamo.c]: Could we add a simple test for this?  I assume it only matters on detach so a test where the app writes after the detach, which we probably already have but it's just not run with logging let alone -log_to_stderr.  Maybe it should just be in the long suite and not the CI suite.

Comment (Carrotman42) [reply core/dynamo.c]: In this file there is already a check to verify `main_logfile != STDERR` before closing it. Is there elsewhere in the codebase that this needs to happen? I looked at all calls to `close_log_file` in the tree and I couldn't find any relevant ones.

Comment (derekbruening) [reply core/dynamo.c]: Oh, ok, did not see that the main logfile is already checked.

Comment (Carrotman42) [reply core/dynamo.c]: Can you advise on how to add a long suite test? I've looked through the cmake config and discovered some parts of the process, but I cannot for the life of me figure out where the test goes or how to mark it as LONG.

Comment (derekbruening) [reply core/dynamo.c]: https://github.com/DynamoRIO/dynamorio/blob/master/suite/tests/CMakeLists.txt#L97

**Comment by fhahn** [comment]
run aarch64 tests

File: core/dynamo.c (hunk 1 above)
Comment (derekbruening) [reply core/dynamo.c:2633]: Going through old PR's: let's just get this fix in.

**Review by derekbruening — APPROVED** [review]

**Pull request merged**